### PR TITLE
update scala versions so we can build on JDK11

### PR DESCRIPTION
**Pull request by richardstartin**

### CODE REVIEW

File: dd-java-agent/instrumentation/scala-concurrent/scala-concurrent.gradle [hunk 1]
```diff
@@ -48,10 +48,9 @@ compileSlickTestGroovy {
 }
 
 dependencies {
-  compileOnly group: 'org.scala-lang', name: 'scala-library', version: '2.8.0'
+  // first version that works with the tests and can build on JDK11 (without dependency problems):
+  implementation group: 'org.scala-lang', name: 'scala-library', version: '2.10.7'
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/scala-concurrent/scala-concurrent.gradle:52]: Does implementation have the same behavior as compileOnly? (Specifically not included as a dependency and not included in the jar)

Comment (richardstartin) [reply dd-java-agent/instrumentation/scala-concurrent/scala-concurrent.gradle:52]: `compileOnly` won't work on gradle 7

Comment (bantonsson) [reply dd-java-agent/instrumentation/scala-concurrent/scala-concurrent.gradle]: So what is the equivalent in Gradle `7`? Using `implementation` is not `compileOnly`, and I couldn't find anything saying that `compileOnly` is deprecated in the `6.8` docs https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph

Comment (tylerbenson) [reply dd-java-agent/instrumentation/scala-concurrent/scala-concurrent.gradle]: Based on [this table](https://docs.gradle.org/current/userguide/java_library_plugin.html) I think we want to replace all `compileOnly` declarations with `compileOnlyApi`.

Comment (richardstartin) [reply dd-java-agent/instrumentation/scala-concurrent/scala-concurrent.gradle]: I'll just change it back for now

**Review by tylerbenson — APPROVED** [review]

**Review by mcculls — APPROVED** [review]

**Comment by richardstartin** [comment]
@bantonsson do you have ideas about how to get `lagomTest` running on JDK11?

**Comment by bantonsson** [comment]
@richardstartin Never really used `Lagom`, but I think that you need to go to the `1.5.x` version to be able to build/run on JDK 11. But that probably means rewriting the test as well.

**Pull request merged**